sdk/core/Azure.Core/src/Shared/RetryAfterPollingStrategy.cs

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+#nullable enable
+
+namespace Azure.Core
+{
+    /// <summary>
+    /// Implementation of a <see cref="OperationPollingStrategy"/> of which the interval is from
+    /// retry-after header of service response.
+    /// </summary>
+    internal class RetryAfterPollingStrategy : OperationPollingStrategy
+    {
+        protected const string RetryAfterHeaderName = "Retry-After";
+        protected const string RetryAfterMsHeaderName = "retry-after-ms";
+        protected const string XRetryAfterMsHeaderName = "x-ms-retry-after-ms";
+
+        private TimeSpan lastRetryAfter { get; set; }
```

Comment:
Shall this start with a upper case letter since it is a property?
```suggestion
        private TimeSpan LastRetryAfter { get; set; }
```